pkg/kn/commands/source/apiserver/apiserver_flags.go

```diff
@@ -0,0 +1,67 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiserver
+
+import (
+	"strconv"
+	"strings"
+
+	sources_v1alpha1 "github.com/knative/eventing/pkg/apis/sources/v1alpha1"
+	"github.com/spf13/cobra"
+)
+
+// ApiServerSourceUpdateFlags are flags for create and update a ApiServerSource
+type ApiServerSourceUpdateFlags struct {
+	accountName string
+	mode        string
+	apiversion  string
+	resources   string
+}
+
+// getApiServerResourceArray is to return an array of ApiServerResource from a string
+// a string sample is Event/v1/true,Pod/v2/false
+func (f *ApiServerSourceUpdateFlags) getApiServerResourceArray() []sources_v1alpha1.ApiServerResource {
+	var resourceList []sources_v1alpha1.ApiServerResource
+	for _, r := range strings.Split(f.resources, ",") {
+		version, kind, controller := getValidResource(r)
+		resourceRef := sources_v1alpha1.ApiServerResource{
+			APIVersion: version,
+			Kind:       kind,
+			Controller: controller,
+		}
+		resourceList = append(resourceList, resourceRef)
+	}
+	return resourceList
+}
+
+func getValidResource(resource string) (string, string, bool) {
+	var version, kind string
+	var isController bool
+	parts := strings.Split(resource, "/")
+	if len(parts) == 3 {
+		version = parts[0]
+		kind = parts[1]
+		isController, _ = strconv.ParseBool(parts[2])
```

Comment:
Why are we ignoring errors? [`strconv.ParseBool`](https://golang.org/pkg/strconv/#ParseBool) returns an error if the string passed is not a bool. Should deal with error.

Reply:
Please address ^^^

Reply:
done